pkg/provider/azure_storageaccount.go

```diff
@@ -262,6 +226,31 @@ func (az *Cloud) EnsureStorageAccount(accountOptions *AccountOptions, genAccount
 					return "", "", err
 				}
 			}
+
+			if accountOptions.CreatePrivateEndpoint {
```

Comment:
does this work? since in Line 210, it's already creating storage account before creating private end point.

Reply:
Yes this is intentional. The storage account is created and then the private endpoint is created for that storage account. I think I don't understand the question here.

Reply:
nvm, code lgtm, pls squash all commits, thanks.